backends/cuda/ceed-cuda.h

```diff
@@ -40,13 +41,46 @@ do { \
   } \
 } while (0)
 
+#define CeedChk_Cublas(ceed, x) \
+do { \
+  cublasStatus_t result = x; \
+  if (result != CUBLAS_STATUS_SUCCESS) { \
+    const char *msg = cublasGetErrorName(result); \
+    return CeedError((ceed), (int)result, msg); \
+   } \
+} while (0)
+
 #define QUOTE(...) #__VA_ARGS__
 
-typedef enum {CEED_CUDA_HOST_SYNC,
-              CEED_CUDA_DEVICE_SYNC,
-              CEED_CUDA_BOTH_SYNC,
-              CEED_CUDA_NONE_SYNC
-             } CeedCudaSyncState;
+static const char *cublasGetErrorName(cublasStatus_t error) {
+  switch (error) {
```

Comment:
Its annoying that this error code -> char utility doesn't ship with CUBLAS

Reply:
Agreed. You can use a macro like
```c
#define CASE(name) case name: return #name
```
and use like
```c
CASE(CUBLAS_STATUS_SUCCESS);
CASE(CUBLAS_STATUS_NOT_INITIALIZED);
...
```
but it still sucks. The same thing can go in an array with
```
const char *const cublas_status[] = {
  [CUBLAS_STATUS_SUCCESS] = "CUBLAS_STATUS_SUCCESS",
  ...
};
```
which may be close to what the compiler does with your switch statement. But in any case, it'd be much better provided by cuBLAS.

Reply:
I like that macro better. At least it doesn't eat as much space. Thanks